src/processing/processing.ts

```diff
@@ -158,3 +167,129 @@ export const postProcessApiCallResponse = async (
 
   return goResult.data;
 };
+
+/**
+ * Pre-processes API call parameters based on the provided endpoint's processing specifications.
+ *
+ * @param endpoint The endpoint containing processing specifications.
+ * @param apiCallParameters The parameters to be pre-processed.
+ * @param processingOptions Options to control the async processing behavior like retries and timeouts.
+ *
+ * @returns A promise that resolves to the pre-processed parameters.
+ */
+export const preProcessApiCallParametersV2 = async (
+  endpoint: Endpoint,
+  apiCallParameters: ApiCallParameters,
+  processingOptions: GoAsyncOptions = { retries: 0, totalTimeoutMs: DEFAULT_PROCESSING_TIMEOUT_MS }
+): Promise<PreProcessingV2Response> => {
+  const { preProcessingSpecificationV2 } = endpoint;
```

Comment:
I wonder if it wouldn't be better to simplify these input arguments. Like there isn't really a need for an "endpoint" to be passed in and importing the type from `@api3/ois`, when we could just pass in the `preProcessingSpecificationV2` (or just `preProcessingSpecification` since the function name has the "v2") if that's all we care about.

Passing the full endpoint is less flexible and requires more understanding to know what an `Endpoint` looks like underneath

Reply:
Yeah, I was thinking about that as well but didn't want to pollute the PR with refactoring. But since you pointed it out, let's do it.

Reply:
https://github.com/api3dao/commons/pull/32/commits/9f6c27591a050db0741cff000a1aecdc25afbe46